internal/fusefs/file.go

```diff
@@ -0,0 +1,136 @@
+//go:build fuse
+
+package fusefs
+
+import (
+	"context"
+	"io"
+	"log/slog"
+	"syscall"
+	"time"
+
+	"github.com/hanwen/go-fuse/v2/fs"
+	"github.com/hanwen/go-fuse/v2/fuse"
+	"github.com/javi11/altmount/internal/nzbfilesystem"
+	"github.com/spf13/afero"
+)
+
+// NzbFile represents a file in our FUSE filesystem.
+type NzbFile struct {
+	fs.Inode
+	metadataRemoteFile *nzbfilesystem.MetadataRemoteFile
+	path               string // The full path of this file in the virtual filesystem
+	uid                uint32
+	gid                uint32
+}
+
+var _ = (fs.InodeEmbedder)((*NzbFile)(nil))
+
+// NzbFileHandle implements fs.FileHandle for an opened file.
+type NzbFileHandle struct {
+	aferoFile afero.File // The underlying afero.File opened from metadataRemoteFile
+}
+
+var _ = (fs.FileHandle)((*NzbFileHandle)(nil))
+
+// Getattr implements fs.Inode.Getattr.
+// It retrieves attributes for the file using MetadataRemoteFile.
+func (f *NzbFile) Getattr(ctx context.Context, fh fs.FileHandle, out *fuse.Attr) syscall.Errno {
+	ok, info, err := f.metadataRemoteFile.Stat(ctx, f.path)
+	if err != nil {
+		return syscall.EIO
+	}
+	if !ok {
+		return syscall.ENOENT
+	}
+
+	out.Mode = ToFuseMode(info.Mode())
+	out.Size = uint64(info.Size())
+	out.Mtime = uint64(info.ModTime().Unix())
+	out.Ctime = uint64(info.ModTime().Unix())
+	out.Atime = uint64(info.ModTime().Unix())
+	out.Uid = f.uid
+	out.Gid = f.gid
+
+	return fs.OK
+}
+
+// Open implements fs.Inode.Open.
+// It opens the file using MetadataRemoteFile and returns an NzbFileHandle.
+func (f *NzbFile) Open(ctx context.Context, flags uint32) (fh fs.FileHandle, fuseFlags uint32, errno syscall.Errno) {
+	// Only allow read-only access
+	if flags&syscall.O_ACCMODE != syscall.O_RDONLY {
+		return nil, 0, syscall.EACCES
+	}
+
+	ok, aferoFile, err := f.metadataRemoteFile.OpenFile(ctx, f.path)
```

Comment:
The current filesystem is not prepared to use seek because it works by Range requests, if fuse needs seek you will need to modify it in metadata_remote_file